web-console/knikubevirt/VMIs/VMIs.md

```diff
@@ -0,0 +1,56 @@
+# Surfacing VMIs in the UI
+
+A virtual machine instance(VMI) is the representation of a running virtual machine.
+
+## Use Cases
+
+1. Debugging -  Previously if a user was trying to resolve an issue caused by a VMI that didn't have an owner it would not be possible to discover it in the UI. Including orphaned VMIs indicates to a user that there could be an issue that they need to address.
+
+2. On Bare metal machines installed with Kubernetes or users who install Kubevirt for CNV - On these VMs there are clusters that can be scaled up and down and they create the VMIs (and not the VMs).
+
+## Mockups
+
+In order to avoid duplication and unclutter the UI, we had finalized into an option that presents VMs and VMIs in a single list view.
+We don't surface VMIs in the list if they are owned by a VM.
+
+VMIs and VMs will be distinguished between different color badges. VMI badges color are darker (#002F5D) in order to provide proper contrast for low vision users.
+
+![single list view of VMs and VMIs ](img/VMsListW_VMIsOp2.png)
+
+## Hints
+
+To better educate users about VMIs, we will provide hints, whenever it is appropriate.
+The instance name link will take the user to the instance page (VM/VMI).
+
+![showing hint pattern to help educate users about VMIs](img/Op2_Hint.png)
+
+## VMI's details view
+
+VMI and VM in the list view will lead to the VM and VMI details page.
+The details page link from VMI to VM (or note that there is no managing entity=owner) and from VM to VMI (or note that there is no VMI).
+
+VMI's details page is similar to the VM’s details page and acts in a complementary manner.
+
+The VMI's details page will provide a hint with a general explanation about what VMIs are as well as allow drill down to further details.
+
+![VMI's details page](img/Op1_VMI_DetailsViewPlusHint.png)
+
+In the VMI’s details page, the owner VM will be represented with a link, so users can click and get there and vice versa.
+
+## Adding a link to the VMI from the VM details page
+
+In order to let the user get to the VMI even when there is a VM around it, we added a link in the VM details page.
+
+![VMI's details page showing link to its VMI](img/VM_DetailsPageW_LinkToVMI.png)
+
+## Actions available for the VMI
+
+![VMI's available actions](img/Op1Actions.png)
```

Comment:
In our design we only show VMI if they have no VM, so users will not have access to a VMI details page if it has a VM that mask it.

@matthewcarleton @yfrimanm  do we want to let user access the "VM-like" details page of a VMI that has a VM ( a. it will be a duplication of the VM details view, b. we do not link to it from the list view ) ? 
WDYT ?

Reply:
@yaacov AFAIR we said we want users to interact with VMs, so IMO, we should allow that.

Reply:
@yaacov what happens if I go to the details page of a VM and click on the instance there? Won't I go to the VMI details page?

Reply:
> @yaacov what happens if I go to the details page of a VM and click on the instance there? Won't I go to the VMI details page?

in https://github.com/openshift/console/pull/3949 **currenlty WIP** you will get to an actuall "Virtual Machine Instanse" page, that is a generic page for VMI , not the "Virtual Machine" page, with the dashboards, metrics and events.

Reply:
ok, I'd vote for the same experience regardless of how you get to the VMI (list view as an orphan or via the VM details page.) Does that make sense?

Reply:
> ok, I'd vote for the same experience regardless of how you get to the VMI (list view as an orphan or via the VM details page.) 

I have a technical problem:
the VMI and VM use the same details page, `../ns/<namesapce>/virtualmachines/<name>/` their is no difference between a VM and a VMI, like the list the details page check if we have a vm, vmi or vm+vmi, it renders vm page if we have vm, and only if we do not have a vm it renders the vmi view/actions

> Does that make sense?

sure, but I think the solution will be to remove the link ... or make vm and vmi a different page, e.g. different URL and different breadcrumbs

Reply:
ah I see what you're saying @yaacov So we don't want to surface VMI in the VM b/c it adds no additional value. We only show VMI details pages when the VM isn't present. That makes sense to me. @yfrimanm WDYT?

Reply:
yes, the vm details page is actually "vm+vmi" details page, when collecting the data to show in the page we take all the data we have from the vm + all the data we have from the vmi, and show a combined view of all the data we have. Linking from the vm details page to the vmi details page is actually linking to the same page if It make sense :-)

Reply:
I see the tricky issue here. So I think we might need to cancel that link. WDYT?

Reply:
> I see the tricky issue here. So I think we might need to cancel that link. WDYT?

looks like it should be a link to pod ( like it was in the beginning ) :-)

Reply:
ya I think that aligns with how we present it in the list view.
VMI only present if it exists on its own. Otherwise you interact with the VM. I like that more actually b/c then we would surface just the Pod from the VM. 
@yfrimanm can you update the designs to reflect this?